mantidproject/mantid
# Add back to back exponential parameters in ENGIN-X parameter file

**Description of work.**

Refined the GSAS/Fullprof coeficients for A,B,S as a function of d-spacing (input parameters for back to back exponential convoluted with a Gaussian) for the ENGIN-X instrument in a Parameter.xml file.

In testing it became apparent the parameter S was getting overridden when a peak was added (i.e. the refined result was not being used) - this is unexpected behaviour given instrument scientists are keeping the Parameter.xml up to date for other instruments as well. Methods were added to the IPeakFunction class and exposed to python (via the property handler and fit browser) such that when a new peak is added, the fwhm of a peak is only set if the parameter that relates to the width has not already set.  For fitting functions where the parameter that determines the width isn't specified (IPeakFunction method overridden in the function) the fwhm is always set.

**To test:**
(1) Load in the ENGIN-X data using the script below

```
from mantid.simpleapi import *
Load(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxenginx\instrument\data\cycle_12_4\ENGINX00193749.nxs', 
     OutputWorkspace='193749')
```

(2) Change the log level to Debug
(3) Plot any spectrum, zoom in on a peak and press Fit
(4) Right-click on plot, Select peak type -> BackToBackExponential
(5) Add a peak, in the log you should see that where the parameters have been determined from the contents of the Parameter.xml file the parameter values post unit conversion are output. For example 
`The FitParameter S = 29.6166 after result-unit conversion`
(6) Check these numbers match those in the fit property browser
(7) Plot the guess and check is reasonably well initialised (contrast with master/nightly if in doubt)

Fixes #29879






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
